src/client/QXmppMixManager.h

```diff
@@ -0,0 +1,154 @@
+    QXmppTask<ChannelNodeResult> requestChannelNodes(const QString &channelJid);
+
+    QXmppTask<ConfigurationResult> requestChannelConfiguration(const QString &channelJid);
+    QXmppTask<QXmppClient::EmptyResult> updateChannelConfiguration(const QString &channelJid, QXmppMixConfigItem configuration);
+    Q_SIGNAL void channelConfigurationUpdated(const QString &channelJid, const QXmppMixConfigItem &configuration);
+
+    QXmppTask<InformationResult> requestChannelInformation(const QString &channelJid);
+    QXmppTask<QXmppClient::EmptyResult> updateChannelInformation(const QString &channelJid, QXmppMixInfoItem information);
+    Q_SIGNAL void channelInformationUpdated(const QString &channelJid, const QXmppMixInfoItem &information);
+
+    QXmppTask<JoiningResult> joinChannel(const QString &channelJid, const QString &nickname = {}, QXmppMixConfigItem::Nodes nodes = ~QXmppMixConfigItem::Nodes());
+    QXmppTask<JoiningResult> joinChannel(const QXmppMixInvitation &invitation, const QString &nickname = {}, QXmppMixConfigItem::Nodes nodes = ~QXmppMixConfigItem::Nodes());
+
+    QXmppTask<NicknameResult> updateNickname(const QString &channelJid, const QString &nickname);
+    QXmppTask<SubscriptionResult> updateSubscriptions(const QString &channelJid, QXmppMixConfigItem::Nodes subscriptionAdditions = ~QXmppMixConfigItem::Nodes(), QXmppMixConfigItem::Nodes subscriptionRemovals = ~QXmppMixConfigItem::Nodes());
+
+    QXmppTask<InvitationResult> requestInvitation(const QString &channelJid, const QString &inviteeJid);
+
+    QXmppTask<JidResult> requestAllowedJids(const QString &channelJid);
+    QXmppTask<QXmppClient::EmptyResult> allowJid(const QString &channelJid, const QString &jid);
+    Q_SIGNAL void jidAllowed(const QString &channelJid, const QString &jid);
+    Q_SIGNAL void allJidsAllowed(const QString &channelJid);
+
+    QXmppTask<QXmppClient::EmptyResult> disallowJid(const QString &channelJid, const QString &jid);
+    Q_SIGNAL void jidDisallowed(const QString &channelJid, const QString &jid);
+    QXmppTask<QXmppClient::EmptyResult> disallowAllJids(const QString &channelJid);
+    Q_SIGNAL void allJidsDisallowed(const QString &channelJid);
+
+    QXmppTask<JidResult> requestBannedJids(const QString &channelJid);
+    QXmppTask<QXmppClient::EmptyResult> banJid(const QString &channelJid, const QString &jid);
+    Q_SIGNAL void jidBanned(const QString &channelJid, const QString &jid);
+
+    QXmppTask<QXmppClient::EmptyResult> unbanJid(const QString &channelJid, const QString &jid);
+    Q_SIGNAL void jidUnbanned(const QString &channelJid, const QString &jid);
+    QXmppTask<QXmppClient::EmptyResult> unbanAllJids(const QString &channelJid);
+    Q_SIGNAL void allJidsUnbanned(const QString &channelJid);
+
+    QXmppTask<ParticipantResult> requestParticipants(const QString &channelJid);
+    Q_SIGNAL void userJoinedOrParticipantModified(const QString &channelJid, const QXmppMixParticipantItem &participantItem);
```

Comment:
`userJoinedOrParticipantModified`:
* `or` in names is an anti-pattern
* there's an inconsistency here: _user_ joined / _participant_ modified.

I'd suggest something like `participantUpdated()`.